js-challenge/weather-api/index.js

```diff
@@ -0,0 +1,47 @@
+window.onload = function () {
+  const APPID = "4b5774e9f3d2a07b84f0f2f88e486224";
+  const city = document.getElementById("city");
+
+  //ページ読み込み時に天気を表示する
+  execute();
+
+  //リストボックス変更時に天気を書き換える
+  document.getElementById("city").addEventListener("change", execute, false);
+
+  /**
+   * 天気情報を取得してhtmlの書き換えを行います
+   */
+  async function execute() {
+    const newWeatherText = await fetchWeather();
```

Comment:
info

`fetchWeather`についてです。

こちらの関数ですが、
- 正常系の場合はString型で天気のテキストを返す
- 異常系（例外のブロックにはいったとき）はnullを返す

という動作をします。
（明示的にreturnを書かないとnullがかえる）

今回、後続の処理にnullを渡してもエラーは起きないですが
後段の処理を行うまえにnullチェックをしておくと安全です

```suggestion
    const newWeatherText = await fetchWeather();
    if(newWeatherText) {
      changeWetherText(newWeatherText);
    }
```

Reply:
info

左辺と右辺は等価になるよう記述できると良いです。

現状のコード
左辺：newWeatherText（新しい天気のテキスト、テキストって書いてあるのでString型かな？と思う）
右辺：fetchWeather（天気をAPI経由で取得する、fetchは外部からのデータ取得によく使われる）

左辺は「文字列」、右辺は「天気というオブジェクト」を取得しそうだな、と読めます。
左辺と右辺の扱っているものが異なるように読めてしまうので、混乱のもとです。

今回下記のようにするのがbetterです。
```suggestion
    const newWeatherText = await fetchWeatherText();
```